### PR TITLE
Add new method to ImageDownloaderDelegate - didFinishDownloadingImageForURL:with:error

### PR DESCRIPTION
**Pull request by tomaskraina**
This allows using network activity indicator correctly. Fixes https://github.com/onevcat/Kingfisher/issues/605

I have been using this my fork for some time and thought it would be good idea to merge it back to the official repo.

Maybe the naming could be better. Any ideas?

### CODE REVIEW

**Comment by onevcat** [comment]
@tomaskraina Thanks for it! Happened to be quite busy recently but this addition looks fine to me at the first glance. I'd take a detail look at it later and get you back then!

**Comment by tomaskraina** [comment]
Thanks!

File: Demo/Demo/Kingfisher-Demo/ViewController.swift [hunk 1]
```diff
@@ -95,3 +95,15 @@ extension ViewController: UICollectionViewDataSourcePrefetching {
         ImagePrefetcher(urls: urls).start()
     }
 }
+
+
+// Inspired by: https://fdp.io/blog/2018/03/22/supporting-compactmap-in-swift-4/
```
Comment (nanoxd) [inline Demo/Demo/Kingfisher-Demo/ViewController.swift]: 👋 It's funny to see my blog post on a library I use 😄

Comment (onevcat) [reply Demo/Demo/Kingfisher-Demo/ViewController.swift]: XD

**Pull request merged**